tests/query/test_query.py

```diff
@@ -77,3 +80,37 @@ def test_query_experiments() -> None:
 
     query.set_experiments({"optimization3": 0.5})
     assert query.get_experiments() == {"optimization3": 0.5}
+
+
+@patch("snuba.query.replace")
+def test_query_transformation(mock_replace: MagicMock) -> None:
```

Comment:
Why are there so many mocks in this test? It seems like you are testing the implementation rather than the desired behavior. All the changes in this PR are local to this repo. You should be able to test your change without any mocks